src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/ParamsCollectionTests.vb

```diff
@@ -0,0 +1,98 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
+
+    Public Class ParamsCollectionTests
+        Inherits BasicTestBase
+
+        <Fact>
+        Public Sub Consume_01()
+
+            Dim csSource =
+"
+public class C1
+{
+    public static void M1(params System.Collections.Generic.IEnumerable<int> a) {}
+    public static void M2(System.Collections.Generic.IEnumerable<int> a) {}
+}
+"
+
+            Dim csCompilation = CreateCSharpCompilation(csSource,
+                                           parseOptions:=CSharp.CSharpParseOptions.Default.WithLanguageVersion(CSharp.LanguageVersion.Preview),
+                                           referencedAssemblies:=TargetFrameworkUtil.GetReferences(TargetFramework.Standard)).EmitToImageReference()
+
+            Dim source1 =
+<compilation>
+    <file name="c.vb"><![CDATA[
+Public Class C
+    Shared Sub Main()
+        C1.M1(1, 2, 3)
+        C1.M1({1, 2, 3})
+
+        C1.M2({1, 2, 3})
+    End Sub
+End Class
+]]></file>
+</compilation>
+
+            Dim comp1 = CreateCompilation(source1, targetFramework:=TargetFramework.Standard, references:={csCompilation}, options:=TestOptions.DebugExe)
+
+            ' PROTOTYPE(ParamsCollections): It looks like we will be much safer to use a different attribute for non-array params collections in metadata.
+            '                               The old VB compiler will not be able to consume them decorated with ParamArrayAttribute neither in normal, nor in expanded form.
+            '                               Therefore, an addition of 'params' modifier is likely to be break VB consumers, and very likely consumers from other languages.
+            '                               We possibly could fix up the new version of VB compiler, but I am not sure it would be worth the effort since we can
+            '                               simply use a different attribute, which is likely to work better for other languages too.
```

Comment:
General question: would a different constructor work as well? IE, if we added a `ParamArray(int version)` constructor, could we avoid having to add an entire new attribute?

Reply:
> General question: would a different constructor work as well?

It would work for VB, I think. But might not work for other compilers/tools.

Reply:
BTW, we can synthesize an attribute type when it is not available, but we cannot inject a constructor to an existing type.

Reply:
True, but I don't expect us to synthesize this attribute. In .NET 8, we tried to move most of the standard embedded attributes into the BCL, and I expect us to continue doing in so in the future. I don't think we should synthesize such an attribute.

Reply:
>  I don't think we should synthesize such an attribute.

I think this is what we have been doing recently for all attributes implicitly applied by compiler (as opposed for attributes applied by user explicitly). We do add the attributes to BCL as well, but do not fail when the type isn't available.

Reply:
> I think this is what we have been doing recently for all attributes implicitly applied by compiler (as opposed for attributes applied by user explicitly)

No, we've been inconsistent about that. Required does not, nor does `ref readonly`.

Reply:
> No, we've been inconsistent about that.

Perhaps. It is quite possible there are outliers. And probably there are good reasons why they are. It is quite possible that the new attribute will fall in to the same category. But, at the moment, that is not obvious to me.

Reply:
I, personally, do not see a down side of synthesizing the attribute type for `params`.